### ISSUE TITLE
2.8.01

### ISSUE THREAD

**Issue by HIGHWAY99** [Eldorados/script.module.urlresolver#280]
v2.8.01 seems to have problems of some kind at the moment.  TZ said he was having troubles with 1ch / urlresolver today after he got back.  I seem to be able to get by one what hosts i use with 1ch, but then I found that the functionality of the code below stopped, and wasn't even making .pyo files to show that it was checkeding them.  Been using this code in some addons to make use of a 2ndary plugins path for faster plugin patches.  Man does Helix ever love to say update failed too.  Some where in the 2.8.0 to 2.8.01 seems to have done something to cause an issue with this and maybe more.

def importURLResolver():
	global urlresolver
	import urlresolver
	_plugin_path=xbmc.translatePath(os.path.join(_addonPath,'resources','lib','plugins'))
	urlresolver.plugnplay.plugin_dirs=[]
	urlresolver.plugnplay.set_plugin_dirs(urlresolver.common.plugins_path,_plugin_path)

**Comment by Eldorados** [comment]
I'm not reading you right, what are the issues?‎Which pyo files aren't being generated? They are just compiled versions of the py's, and not all get compiled2.8.01 is just beta at least so is not out to the masses yet, I have another beta I'll be pushing out on Monday                                                                                                                                                                                                                                                                                                                                         Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: HIGHWAY99Sent: Saturday, March 21, 2015 6:43 PMTo: Eldorados/script.module.urlresolverReply To: Eldorados/script.module.urlresolverSubject: [script.module.urlresolver] 2.8.01 (#280)v2.8.01 seems to have problems of some kind at the moment.  TZ said he was having troubles with 1ch / urlresolver today after he got back.  I seem to be able to get by one what hosts i use with 1ch, but then I found that the functionality of the code below stopped, and wasn't even making .pyo files to show that it was checkeding them.  Been using this code in some addons to make use of a 2ndary plugins path for faster plugin patches.  Man does Helix ever love to say update failed too.  Some where in the 2.8.0 to 2.8.01 seems to have done something to cause an issue with this and maybe more.

def importURLResolver():
    global urlresolver
    import urlresolver
    _plugin_path=xbmc.translatePath(os.path.join(_addonPath,'resources','lib','plugins'))
    urlresolver.plugnplay.plugin_dirs=[]
    urlresolver.plugnplay.set_plugin_dirs(urlresolver.common.plugins_path,_plugin_path)

—Reply to this email directly or view it on GitHub.

**Comment by HIGHWAY99** [comment]
i've tried to the mar16th pull requests to see if anything wrong happened in any of them and nothing so far.

Anime44, AnimeGet, and other addons are using the code mentioned before there to use a 2nd plugins folder along with the default one to hold some extra plugins.  Something done in 2.8.01 that wasn't in 2.8.0 seems to have broken the ability to use that.

**Comment by HIGHWAY99** [comment]
Am not seeing a comment that looks like anything would have been done to effect this, so thought you might have some idea, with knowing better than me whats been changed.

I've tried cleaning the script.module.urlresolver folder out and extracting from 2.8.0's zip and then 2.8.0.1's zip again and back and furth to pin down that it was coming from something in url resolver.

i know \resolver\__init__.py has 
plugnplay.set_plugin_dirs(common.plugins_path)
and \resolver\plugnplay\__init__.py has 
def set_plugin_dirs(*dirs):
  for d in dirs:
    common.addon.log_debug('adding plugin dir: %s' % d)
    plugin_dirs.append(d)  


Also wasn't sure if it might be the sake of a host plugin in the list before it having an error maybe?! causing it to not look at the rest.

I've tried with custom folder last (takes priority it seems), first as we originally did, and both first and last:

	urlresolver.plugnplay.set_plugin_dirs(urlresolver.common.plugins_path,_plugin_path)
	urlresolver.plugnplay.set_plugin_dirs(_plugin_path,urlresolver.common.plugins_path,_plugin_path)
	urlresolver.plugnplay.set_plugin_dirs(_plugin_path,urlresolver.common.plugins_path)

**Comment by Eldorados** [comment]
I'm still lost at what the actual issue is here, what is failing? Error msgs?                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: HIGHWAY99Sent: Saturday, March 21, 2015 7:07 PMTo: Eldorados/script.module.urlresolverReply To: Eldorados/script.module.urlresolverCc: EldoradosSubject: Re: [script.module.urlresolver] 2.8.01 (#280)Am not seeing a comment that looks like anything would have been done to effect this, so thought you might have some idea, with knowing better than me whats been changed.

I've tried cleaning the script.module.urlresolver folder out and extracting from 2.8.0's zip and then 2.8.0.1's zip again and back and furth to pin down that it was coming from something in url resolver.

i know \resolver__init__.py has 
plugnplay.set_plugin_dirs(common.plugins_path)
and \resolver\plugnplay__init__.py has 
def set_plugin_dirs(*dirs):
  for d in dirs:
    common.addon.log_debug('adding plugin dir: %s' % d)
    plugin_dirs.append(d)  

Also wasn't sure if it might be the sake of a host plugin in the list before it having an error maybe?! causing it to not look at the rest.

I've tried with custom folder last (takes priority it seems), first as we originally did, and both first and last:

urlresolver.plugnplay.set_plugin_dirs(urlresolver.common.plugins_path,_plugin_path)
urlresolver.plugnplay.set_plugin_dirs(_plugin_path,urlresolver.common.plugins_path,_plugin_path)
urlresolver.plugnplay.set_plugin_dirs(_plugin_path,urlresolver.common.plugins_path)


—Reply to this email directly or view it on GitHub.

**Comment by HIGHWAY99** [comment]
error message is the hard part, as it's returning False as the media_url but other than that i'm not really getting any error message.

Hmm nevermind man i think i figured out something that smokdpi told me before that we didn't end up needing on 2.8.0 but seems to now need on 2.8.0.1.

urlresolver.plugnplay.load_plugins()
used after the other code seems to let it work again. wow, am i ever glad i remembered finally that there was that line that didn't seem to be needed before.

basically it didn't seem to be loading the plugins from the 2nd location anymore in 2.8.0.1, but was in 2.8.0 without that command. lol

so no biggy i think
